homeassistant/components/axis.py

```diff
@@ -143,16 +148,18 @@ def configuration_callback(callback_data):
     )
 
 
-def setup(hass, config):
+@asyncio.coroutine
+def async_setup(hass, config):
     """Common setup for Axis devices."""
     def _shutdown(call):  # pylint: disable=unused-argument
-        """Stop the metadatastream on shutdown."""
+        """Stop the event stream on shutdown."""
         for serialnumber, device in AXIS_DEVICES.items():
-            _LOGGER.info("Stopping metadatastream for %s.", serialnumber)
-            device.stop_metadatastream()
+            _LOGGER.info("Stopping event stream for %s.", serialnumber)
+            device.stop()
```

Comment:
are these methods async? (because this is a sync context as the shutdown function has not been annotated)

Reply:
Do I need to decorate with coroutine in all sub functions for it to be a proper async context?

Reply:
Decorated with callback is that OK?